### PR TITLE
Document more tools

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
There are many other cool tools and utils that people may find useful, lets tell them about it.

##### ISSUE TYPE
 - Docs Pull Request

### CODE REVIEW

**Review by bcoca — APPROVED** [review]

File: docs/docsite/rst/community/other_tools_and_programs.rst [hunk 1]
```diff
@@ -1,24 +1,108 @@
...
+
+* `Language YAML <https://github.com/atom/language-yaml>`_
+
+
+Pycharm
+=======
+
+Is a full IDE to develop Python software. It has all the needed parts to be able to write
+python scripts and complete software. Is a little overkill for writing roles/playbooks but, it can
+be a very useful tool if you write modules and submit code for Ansible.
```
Comment (jborean93) [inline docs/docsite/rst/community/other_tools_and_programs.rst]: Is it worth adding a comment saying you can use PyCharm to debug the engine compontents or is designed to be module specific?

File: docs/docsite/rst/community/other_tools_and_programs.rst [hunk 2]
```diff
@@ -1,24 +1,108 @@
...
+* `Ansible vim <https://github.com/pearofducks/ansible-vim>`_  - vim syntax plugin for Ansible 2.x, it supports YAML playbooks, Jinja2 templates, and Ansible's hosts files.
 
-- `Ansible Lint <https://github.com/willthames/ansible-lint>`_ is a widely used, highly configurable best-practices linter for Ansible playbooks.
 
+********************
+Validating Playbooks
+********************
+
+- `Ansible Lint <https://github.com/willthames/ansible-lint>`_ is a widely used, highly configurable best-practices linter for Ansible playbooks.
 - `Ansible Review <http://willthames.github.io/2016/06/28/announcing-ansible-review.html>`_ is an extension of Ansible Lint designed for code review.
```
Comment (jborean93) [inline docs/docsite/rst/community/other_tools_and_programs.rst]: Probably good to change the URL to point to the repo which aligns with the new entries you've added https://github.com/willthames/ansible-review

File: docs/docsite/rst/community/other_tools_and_programs.rst (hunk 1 above)
Comment (gundalow) [reply docs/docsite/rst/community/other_tools_and_programs.rst]: Meant to be generic, good call. Updated.

**Review by jborean93 — APPROVED** [review]

**Comment by gundalow** [comment]
Thanks for the tech reviews, with @acozine for wordsmithing. 
This should be backported to `stable-2.6`

**Review by maxamillion — APPROVED** [review]

**Comment by maxamillion** [comment]
shipit

**Review by acozine — CHANGES_REQUESTED** [review]
I reorganized, tidied, and generally word-smithed this content in this commit: https://github.com/acozine/ansible/commit/90b3a640ee36f40ece2a777d1d63309268bad1be (the branch is `pr43053`). I moved ARA and Ansible Inventory Grapher in the Other Tools section. Otherwise my suggested changes are unlikely to be controversial. I can't push my branch to @gundalow's repo, though.

**Review by acozine — APPROVED** [review]

**Comment by jctanner** [comment]
bot_status

**Pull request merged**